### PR TITLE
[Refactoring] Simplify in doc.c

### PR DESCRIPTION
**Pull request by 9rnsr**
Points:
1. In `emitComment` and `emitDitto`, `highlightCode` is unnecessarily conditionally called. Remove the conditions will make code flow simpler.
2. Change eponymous template handling to be consistent.

Additionally fix code indentation.

### CODE REVIEW

**Comment by 9rnsr** [comment]
Required by #4632 and further ddoc improvements - I'm working on [enhancement 13270](https://issues.dlang.org/show_bug.cgi?id=13270).

File: src/doc.c [hunk 1]
```diff
@@ -1822,7 +1858,8 @@ void DocComment::parseMacros(Escape **pescapetable, Macro **pmacrotable, const u
         }
 
         if (*p != '=')
-        {   if (namelen)
+        {
```
Comment (andralex) [inline src/doc.c]: muahaha thanks

File: src/doc.c [hunk 2]
```diff
@@ -951,15 +993,21 @@ void emitComment(Dsymbol *s, Scope *sc)
...
+                size_t o = buf->offset;
+                toDocBuffer(em, buf, sc);
+                highlightCode(sc, em, buf, o);
+                sc->lastoffset = buf->offset;
+            }
             buf->writestring(ddoc_decl_e);
 
             buf->writestring(ddoc_decl_dd_s);
-            dc->writeSections(sc, em, buf);
+            {
```
Comment (andralex) [inline src/doc.c]: why is this scope necessary?

File: src/doc.c [hunk 3]
```diff
@@ -697,47 +719,52 @@ static void expandTemplateMixinComments(TemplateMixin *tm, Scope *sc)
...
+    if (sds->isModule())
+        m = "$(DDOC_MODULE_MEMBERS ";
+    else if (sds->isClassDeclaration())
+        m = "$(DDOC_CLASS_MEMBERS ";
+    else if (sds->isStructDeclaration())
+        m = "$(DDOC_STRUCT_MEMBERS ";
+    else if (sds->isEnumDeclaration())
+        m = "$(DDOC_ENUM_MEMBERS ";
+    else if (sds->isTemplateDeclaration())
+        m = "$(DDOC_TEMPLATE_MEMBERS ";
```
Comment (andralex) [inline src/doc.c]: just a note on efficiency, this should be ordered in order of decreasing likelihood, i.e. struct and template should come first and module should come last etc.

File: src/doc.c [hunk 4]
```diff
@@ -792,29 +827,45 @@ void emitComment(Dsymbol *s, Scope *sc)
...
+                toDocBuffer(d, buf, sc);
+                highlightCode(sc, d, buf, o);
+                sc->lastoffset = buf->offset;
+            }
             buf->writestring(ddoc_decl_e);
 
+            // Put the ddoc comment as the document 'description'
             buf->writestring(ddoc_decl_dd_s);
-            dc->writeSections(sc, d, buf);
+            {
```
Comment (andralex) [inline src/doc.c]: why the scope here?

File: src/doc.c [hunk 5]
```diff
@@ -822,16 +873,22 @@ void emitComment(Dsymbol *s, Scope *sc)
...
+                toDocBuffer(ad, buf, sc);
+                highlightCode(sc, ad, buf, o);
+                sc->lastoffset = buf->offset;
+            }
             buf->writestring(ddoc_decl_e);
 
             buf->writestring(ddoc_decl_dd_s);
-            dc->writeSections(sc, ad, buf);
-            emitMemberComments(ad, sc);
+            {
```
Comment (andralex) [inline src/doc.c]: same question (I understand when scoping is used for defining locals inside, but some of these do not)

File: src/doc.c [hunk 6]
```diff
@@ -840,55 +897,38 @@ void emitComment(Dsymbol *s, Scope *sc)
...
-            sc->lastoffset = buf->offset;
+                sc->lastoffset = buf->offset;
+            }
             buf->writestring(ddoc_decl_e);
 
             buf->writestring(ddoc_decl_dd_s);
-            dc->writeSections(sc, td, buf);
-            if (hasmembers)
-                emitMemberComments((ScopeDsymbol *)ss, sc);
+            {
```
Comment (andralex) [inline src/doc.c]: ++

File: src/doc.c [hunk 7]
```diff
@@ -920,16 +958,22 @@ void emitComment(Dsymbol *s, Scope *sc)
...
+                toDocBuffer(ed, buf, sc);
+                highlightCode(sc, ed, buf, o);
+                sc->lastoffset = buf->offset;
+            }
             buf->writestring(ddoc_decl_e);
 
             buf->writestring(ddoc_decl_dd_s);
-            dc->writeSections(sc, ed, buf);
-            emitMemberComments(ed, sc);
+            {
```
Comment (andralex) [inline src/doc.c]: ++

File: src/doc.c (hunk 2 above)
Comment (9rnsr) [inline src/doc.c]: By adding indentation, dmd newbie will be able to guess the ddoc output.

File: src/doc.c [hunk 8]
```diff
@@ -1206,142 +1250,111 @@ void toDocBuffer(Dsymbol *s, OutBuffer *buf, Scope *sc)
...
-                    visit((Declaration *)fd);
-                }
-            }
-        }
-
         void visit(AggregateDeclaration *ad)
         {
-            if (ad->ident)
-            {
+            if (!ad->ident)
```
Comment (andralex) [inline src/doc.c]: love it

File: src/doc.c (hunk 3 above)
Comment (9rnsr) [inline src/doc.c]: OK will do.

File: src/doc.c (hunk 4 above)
Comment (9rnsr) [inline src/doc.c]: For the code readability.

**Comment by andralex** [comment]
lgtm according to my sketchy understanding of the code involved

File: src/doc.c (hunk 3 above)
Comment (9rnsr) [inline src/doc.c]: Done.

File: src/doc.c (hunk 5 above)
Comment (WalterBright) [inline src/doc.c]: This is the 'new coding style' I referred to.

**Comment by 9rnsr** [comment]
@WalterBright I removed local scope additions that you don't like. Is this OK to merge?

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
Thank you, Kenji!

**Comment by 9rnsr** [comment]
Thanks for toggling auto-merge, but currently it doesn't work because it's blocked by the random druntime failure...

**Comment by CyberShadow** [comment]
This pull request introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=14594

**Pull request merged**